### PR TITLE
Add primary & secondary names

### PR DESCRIPTION
**Pull request by cammellos**
adds primary secondary name, and a bunch of other fields that are used in mobile frontend.

### CODE REVIEW

File: protocol/contact.go [hunk 1]
```diff
@@ -191,10 +179,45 @@ func (c *Contact) mutual() bool {
...
+func (c *Contact) PrimaryName() string {
+	return c.names()[0]
+}
+
+func (c *Contact) SecondaryName() string {
+	names := c.names()
+	if len(names) > 1 {
+		return names[1]
+	}
+	return ""
```
Comment (flexsurfer) [inline protocol/contact.go:222]: can we return nil ?

**Review by flexsurfer — APPROVED** [review]

File: protocol/contact.go [hunk 2]
```diff
@@ -191,10 +179,45 @@ func (c *Contact) mutual() bool {
...
 }
 
+func (c *Contact) names() []string {
+	var names []string
+
+	if c.LocalNickname != "" {
+		names = append(names, c.LocalNickname)
+	}
+
+	if c.ENSVerified && len(c.EnsName) != 0 {
```
Comment (flexsurfer) [inline protocol/contact.go:197]: hey @cammellos sorry i didn't review properly, we should have secondary name only if we have nickname, if we don't have it, secondary name should be empty, so there is no need in names array, we should set only two fields, 
```clojure
(defn assoc-names
  [{:keys [nickname display-name ens-name ens-verified alias public-key] :as contact}]
  (let [username (cond ens-verified
                       ens-name
                       (not (string/blank? display-name))
                       display-name
                       :else
                       (or alias (gfycat/generate-gfy public-key)))]
    (if (not (string/blank? nickname))
      (assoc contact :first-name nickname :second-name username)
      (assoc contact :first-name username))))
```

**Review by flexsurfer — CHANGES_REQUESTED** [review]

File: protocol/contact.go (hunk 2 above)
Comment (cammellos) [reply protocol/contact.go:197]: oh I see, I'll change that then

**Review by MishkaRogachev — APPROVED** [review]
LGTM!

**Review by flexsurfer — APPROVED** [review]

**Review by alwx — APPROVED** [review]

**Pull request merged**